File: 1. Two Sum/two-sum.go
// 1. Two Sum (11/16/2021)
// Runtime: 4 ms (88.83%) Memory: 4.34 MB (21.83%) 

func twoSum(nums []int, target int) []int {
    table := make(map[int]int)
    
    for i := 0; i < len(nums); i++ {
        if val, exists := table[nums[i]]; exists {
            return []int{val, i}
            break
        }
        table[target - nums[i]] = i
    }
    return []int{0, 1}
}

File: 1. Two Sum/two-sum.js
// 1. Two Sum (3/14/53371)
// Runtime: 72 ms (82.97%) Memory: 39.42 MB (94.91%) 

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = function(nums, target) {
    let table = {};
    
    for(var i = 0; i < nums.length; i++) {
        if(table[nums[i]] >= 0) {
            return [table[nums[i]], i];
        }
        
        table[target - nums[i]] = i;
    }
};